src/Components/Server/src/Circuits/CircuitOptionsJSInteropDetailedErrorsConfiguration.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    internal class CircuitOptionsJSInteropDetailedErrorsConfiguration : IConfigureOptions<CircuitOptions>
+    {
+        public CircuitOptionsJSInteropDetailedErrorsConfiguration(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        public void Configure(CircuitOptions options)
+        {
+            options.JSInteropDetailedErrors = Configuration.GetValue<bool>(WebHostDefaults.DetailedErrorsKey);
```

Comment:
Previously when we talked about using `WebHostDefaults.DetailedErrorsKey`, I might have misunderstood the proposal a bit. I thought we'd be doing the same as `GenericWebHostedService` which shows detailed errors if the configured value is `true` **or** if `HostingEnvironment.IsDevelopment()`: see https://github.com/aspnet/AspNetCore/blob/master/src/Hosting/Hosting/src/GenericHost/GenericWebHostedService.cs#L154

Unless I'm misunderstanding, it looks like our logic here would be inconsistent with that, in that it would be based only on the configured value and would ignore the hosting environment. Do you think this could be made to respect the hosting environment too?

Most newcomers to ASP.NET Core won't immediately have any idea how to configure this, so hiding exceptions from them in development would make it hard to start using the system.

Reply:
Everything else in the PR looks good to me though!

Reply:
I can add that, but as I mentioned in my previous comments, @davidfowl and Damian have been very vocal against doing that (having the framework key features/functionality based on .IsDevelopment()) so I want to get @davidfowl opinion on it.

Maybe their thoughts in the matter have changed, but we wouldn't go to the trouble of having ifdefs in the template if that was not a concern anymore.

Reply:
AFAIK the only two places where its ok to use IsDevelopment are within hosting and within the metapackage. 
In other places of the framework the recommendation is to use config.
We can put it in `launchSettings.json` in the template to achieve the same effect.

Reply:
By all means let's check with @davidfowl / @DamianEdwards, but what I'm suggesting isn't a new design - it's just continuing the existing design we have elsewhere.

We could change that design across the whole stack, but it seems weirder to add new things that are inconsistent with it, especially when both can be involved in the same app at the same time. For example you could have page exceptions giving detailed errors, but JS interop exceptions hiding detailed errors, within the same running instance of an app, which seems super disorganized.

Reply:
@SteveSandersonMS I understand your concern and that's why I suggested we had a unified view of the concept.

My general understanding is that the guidelines that we follow are:
* Metapackage can use IsDevelopment.
* Hosting/Servers can use IsDevelopment.
* Anything else in the framework should use a different mechanism, like config.

My proposal is that we merge the PR as it is right now and follow up on a conclusion to this, as it is a trivial change. I'll start a thread if you want and once we get to a resolution I'll send a PR to add the IsDevelopment check which should be trivial.

Does that sound good?

Reply:
Sure, as long as some follow-up discussion happens, that sounds good to me.

I also like your idea of enabling the detailed errors in `appsettings.Development.json`.

Reply:
I'll make sure it happens.